ansible_collections/arista/avd/docs/input-variable-validation-BETA.md

```diff
@@ -0,0 +1,211 @@
+# The converted data is set as facts which can be seen with -v, but is more readable with -vvv
+avd_validate_conversion_mode: < "disabled" | "warning" | "message" | "debug" | default -> "debug" >
+
+# Validation Mode for arista.avd.validate | Optional
+# During validation, messages will generated with information about the host(s) and key(s) which failed validation
+# "disabled" means that validation will not run.
+# "error" will produce error messages and fail the task.
+# "warning" will produce warning messages
+# "message" will produce regular messages
+# "debug" will produce hidden messages viewable with -v
+avd_validate validation_mode: < "disabled" | "error" | "warning" | "message" | "debug" | default -> "warning" >
+```
+
+## Schema Details
+
+The AVD Schema format is a proprietary schema, inspired from JSON Shema, Ansible Argument Specificion and Arista
+Cloudvision Studios formats.
+
+The AVD Schema have special features which are not supported or possible with any of the mentioned schema formats.
+Schema converters can be developed as needed.
+
+Here is an example of a schema fragment for `bgp_as`:
+
+```yaml
+type: dict
+allow_other_keys: true
+keys:
+  bgp_as:
+    display_name: BGP AS
+    description: AS number to use to configure overlay when "overlay_routing_protocol" == IBGP
+    type: str
+    convert_types:
+      - int
+```
+
+This fragment will be merged with other fragments during development, to form the complete role schema.
+
+For reference the full Role Schemas can be found here:
+- [`eos_designs` AVD Schema](../roles/eos_designs/schemas/eos_designs.schema.yml)
+- [`eos_cli_config_gen` AVD Schema](../roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml)
+
+The supported schema options depends of the type of variable that is described. The supported types are `int`, `bool`, `str`,
+`dict` and `list`. The schema does not support mixed types for the same variable, but the automatic type conversion mentioned
+above, can address the usability aspect by helping the user with common mistakes.
+
+The supported schema options for AVD Schema are described in a meta-schema using JSON Schema Draft-7 format. The meta-schema
+can be seen [here](../plugins/plugin_utils/schema/avd_meta_schema.json). Below is a more detailed description of the supported
+schema options per variable type.
+
+### Schema Options for type `int` (Integer)
+
+| Key | Type | Required | Default | Value Restrictions | Description |
+| ----| ---- | -------- | ------- | ------------------ | ----------- |
+| <samp>type</samp> | String | True | | Valid Values:<br>- `"int"` | Type of variable using the Python short names for each type.<br>`int` for Integer |
+| <samp>convert_types</samp> | List | | | | List of types to auto-convert from.<br>For type `int`, auto-conversion is supported from `bool` and `str`" |
+| <samp>&nbsp;&nbsp;- \<str\></samp> | String | | | Valid Values:<br>- `"bool"`<br>- `"str"` | |
```

Comment:
I found out that I forgot to add `List, Items: String` to the line above. Still it makes sense to have a line for the items schema, since this could be anything - ex a dict.